routing/index_graph.hpp

```diff
@@ -0,0 +1,83 @@
+#pragma once
+
+#include "routing/edge_estimator.hpp"
+#include "routing/geometry.hpp"
+#include "routing/joint.hpp"
+#include "routing/joint_index.hpp"
+#include "routing/road_index.hpp"
+#include "routing/road_point.hpp"
+
+#include "coding/reader.hpp"
+#include "coding/write_to_sink.hpp"
+
+#include "geometry/point2d.hpp"
+
+#include "std/cstdint.hpp"
+#include "std/shared_ptr.hpp"
+#include "std/unique_ptr.hpp"
+#include "std/utility.hpp"
+#include "std/vector.hpp"
+
+namespace routing
+{
+class JointEdge final
+{
+public:
+  JointEdge(Joint::Id target, double weight) : m_target(target), m_weight(weight) {}
+  Joint::Id GetTarget() const { return m_target; }
+  double GetWeight() const { return m_weight; }
+
+private:
+  Joint::Id const m_target;
```

Comment:
| m_target| could be beginning or ending of edge (not only ending). It depends on using of this class. Let's add a comment about it.